index.go

```diff
@@ -0,0 +1,275 @@
+		acceptedStatusCodes: []int{http.StatusNoContent},
+		functionName:        "Delete",
+	}
+	// err is not nil if status code is not 204 StatusNoContent
+	if err := i.client.executeRequest(req); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func (i Index) GetStats() (resp *StatsIndex, err error) {
+	resp = &StatsIndex{}
+	req := internalRequest{
+		endpoint:            "/indexes/" + i.UID + "/stats",
+		method:              http.MethodGet,
+		withRequest:         nil,
+		withResponse:        resp,
+		acceptedStatusCodes: []int{http.StatusOK},
+		functionName:        "GetStats",
+	}
+	if err := i.client.executeRequest(req); err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+func (i Index) GetUpdateStatus(updateID int64) (resp *Update, err error) {
+	resp = &Update{}
+	req := internalRequest{
+		endpoint:            "/indexes/" + i.UID + "/updates/" + strconv.FormatInt(updateID, 10),
+		method:              http.MethodGet,
+		withRequest:         nil,
+		withResponse:        resp,
+		acceptedStatusCodes: []int{http.StatusOK},
+		functionName:        "GetUpdateStatus",
+	}
+	if err := i.client.executeRequest(req); err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+func (i Index) GetAllUpdateStatus() (resp []Update, err error) {
+	resp = []Update{}
+	req := internalRequest{
+		endpoint:            "/indexes/" + i.UID + "/updates",
+		method:              http.MethodGet,
+		withRequest:         nil,
+		withResponse:        &resp,
+		acceptedStatusCodes: []int{http.StatusOK},
+		functionName:        "GetAllUpdateStatus",
+	}
+	if err := i.client.executeRequest(req); err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+// DefaultWaitForPendingUpdate checks each 50ms the status of a WaitForPendingUpdate.
+// This is a default implementation of WaitForPendingUpdate.
+func (i Index) DefaultWaitForPendingUpdate(updateID *AsyncUpdateID) (UpdateStatus, error) {
+	ctx, cancelFunc := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancelFunc()
+	return i.WaitForPendingUpdate(ctx, time.Millisecond*50, updateID)
+}
+
+// WaitForPendingUpdate waits for the end of an update.
+// The function will check by regular interval provided in parameter interval
+// the UpdateStatus. If it is not UpdateStatusEnqueued or the ctx cancelled
+// we return the UpdateStatus.
+func (i Index) WaitForPendingUpdate(
+	ctx context.Context,
+	interval time.Duration,
+	updateID *AsyncUpdateID) (UpdateStatus, error) {
+	for {
+		if err := ctx.Err(); err != nil {
+			return "", err
+		}
+		update, err := i.GetUpdateStatus(updateID.UpdateID)
+		if err != nil {
+			return UpdateStatusUnknown, nil
+		}
+		if update.Status != UpdateStatusEnqueued {
+			return update.Status, nil
+		}
+		time.Sleep(interval)
+	}
+}
+
+func (i Index) Search(request SearchRequest) (*SearchResponse, error) {
+	resp := &SearchResponse{}
+	searchPostRequestParams := map[string]interface{}{}
+	if request.Limit == 0 {
+		request.Limit = 20
+	}
```

Comment:
I don't understand this... If limit is 0, limit becomes 20?

Reply:
I didn't modify this part of the code. But I think it is in case the limit is not specified in the `searchRequest` parameter the default value of meilisearch will be taken.

Reply:
Oh yes, you are right, this is because of the default zero values in Go. If you don't specify any limit, Go will initialize the struct and make it 0. So we modify it to 20. I remember we had a bug before and this is how it was fixed 👌🏽

Reply:
There will be no way to have a `Limit = 0`, but will someone use this kind of request? That is OK from my POV